main.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type StatusChecker interface {
+	Check(ctx context.Context, name string) (status bool, err error)
+}
+
+type httpChecker struct {
+}
+
+/*
+Function having context working as client to fetch status of websit
+*/
+
+func (h httpChecker) Check(ctx context.Context, sitename string) (status bool,
+	err error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```

Comment:
why do we need this ?

Reply:
It's a timeout to wait for any website upto 1 minute only.
If it doesn't send response then we can mark it as DOWN.